src/main/java/md/jack/task9/Primes.java

```diff
@@ -0,0 +1,151 @@
+ * and find the next prime number above a given BigInteger.
+ * <p>
+ * From <a href="http://courses.coreservlets.com/Course-Materials/">the
+ * coreservlets.com tutorials on JSF 2, PrimeFaces, Ajax, JavaScript, jQuery, GWT, Android,
+ * Spring, Hibernate, JPA, RESTful Web Services, Hadoop, Spring MVC,
+ * servlets, JSP, Java 8 lambdas and streams (for those that know Java already),
+ * and Java 8 programming (for those new to Java)</a>.
+ */
+
+public class Primes
+{
+    private static final BigInteger ZERO = BigInteger.ZERO;
+    private static final BigInteger ONE = BigInteger.ONE;
+    private static final BigInteger TWO = new BigInteger("2");
+
+    // Likelihood of false prime is less than 1/2^ERR_VAL.
+    // Presumably BigInteger uses the Miller-Rabin test or
+    // equivalent, and thus is NOT fooled by Carmichael numbers.
+    // See Cormen et al.'s Introduction to Algorithms for details.
+    private static final int ERR_VAL = 100;
+    private static final String[] DIGITS =
+            "0,1,2,3,4,5,6,7,8,9".split(",");
+    private static final String[] NON_ZERO_DIGITS =
+            "0,1,2,3,4,5,6,7,8,9".split(",");
+
+    private Primes()
+    {
+    } // Uninstantiatable class
+
+    public static boolean isPrime(BigInteger possiblePrime)
+    {
+        return (possiblePrime.isProbablePrime(ERR_VAL));
+    }
+
+    public static boolean isPrime(long possiblePrime)
+    {
+        return (isPrime(new BigInteger(Long.toString(possiblePrime))));
+    }
+
+    /** Finds the next prime number above a threshold. */
+
+    private static BigInteger nextPrime(BigInteger start)
+    {
+        if (isEven(start))
+        {
+            start = start.add(ONE);
+        }
+        else
+        {
+            start = start.add(TWO);
+        }
+        if (isPrime(start))
+        {
+            return (start);
+        }
+        return (nextPrime(start));
+    }
+
+    /**
+     * Generates a random number of the given length,
+     * then finds the first prime number above that random number.
+     */
+    public static BigInteger findPrime(int numDigits)
+    {
+        if (numDigits < 1)
+        {
+            numDigits = 1;
+        }
+        return (nextPrime(randomNum(numDigits)));
+    }
+
+    private static boolean isEven(BigInteger n)
+    {
+        return (n.mod(TWO).equals(ZERO));
+    }
+
+    private static String randomDigit(boolean isZeroOk)
+    {
+        if (isZeroOk)
+        {
+            return (RandomUtils.randomElement(DIGITS));
+        }
+        return (RandomUtils.randomElement(NON_ZERO_DIGITS));
+    }
+
+    /**
+     * Creates a random big integer where every digit is
+     * selected randomly (except that the first digit
+     * cannot be a zero).
+     */
+
+    public static BigInteger randomNum(int numDigits)
+    {
+        StringBuilder s = new StringBuilder();
+        // First digit must be non-zero.
+        s.append(randomDigit(false));
+
+        Stream.of(numDigits)
```

Comment:
How does this stream work ?

Reply:
for (int i = 0; i < numDigits - 1; i++)
        {
            s.append(randomDigit(true));
        }
        return (new BigInteger(s.toString()));
something like this , or that's wrong ?